### PR TITLE
Support PolySlab along x/y axis

### PR DESCRIPTION
**Pull request by weiliangjin2021**

### CODE REVIEW

File: tests/test_sidewall.py [hunk 1]
```diff
@@ -219,7 +221,7 @@ def test_intersection_with_inside():
         angle = np.pi / 4
         # avoid vertex-edge crossing case
         try:
```
Comment (tylerflex) [inline tests/test_sidewall.py:223]: what is the idea behind this `try: except:`?

**Review by tylerflex — CHANGES_REQUESTED** [review]
Looks good overall, thanks for this! See a minor comment about the try except, maybe this leftover from development?

About the polyslab tests, it seems now we randomly select the slab axis in the sidewall tests? I wonder if we could add another test, maybe in `test_components` to ensure that the polyslab axis was applied correctly.

For example:

```py
@pytest.mark.parameterize(axis, [0,1,2])
def test_polyslab_axis(axis):
    ps = td.Polyslab(slab_bounds=(-1, 1), vertices=((-5,-5), (-5, 5), (5,5), (5, -5)), axis=axis)

    # assert that the plane `axis=3, axis=-3` do not intersect with the polyslab.
    # assert that the bounds has size  `[10, 10].insert(3, axis)`
    # assert that the point `[0,0].insert(3, axis)` is not inside the polyslab. 
```
Just to make sure that the geometric primitive is working as expected.
For example, we had a bug a few months ago where I forgot to handle the axis correctly for cylinder's inside function.

File: tests/test_sidewall.py (hunk 1 above)
Comment (weiliangjin2021) [reply tests/test_sidewall.py:223]: The polyslab is generated from random vertices. So some of polyslabs will not pass the validation. There are two main validations: 1) no vertex-vertex crossing during dilation/erosion 2) no vertex-edge crossing during dilation/erosion. 1) can be figured out analytically, but I don't have a way to find out 2) analytically yet. So all vertices generated in this test will satisfy 1), but it's unknown if they satisfy 2). This `try: except:` is to rule out all polyslabs that fails 2).

**Comment by weiliangjin2021** [comment]
> Looks good overall, thanks for this! See a minor comment about the try except, maybe this leftover from development?
> 
> About the polyslab tests, it seems now we randomly select the slab axis in the sidewall tests? I wonder if we could add another test, maybe in `test_components` to ensure that the polyslab axis was applied correctly.
> 
> For example:
> 
> ```python
> @pytest.mark.parameterize(axis, [0,1,2])
> def test_polyslab_axis(axis):
>     ps = td.Polyslab(slab_bounds=(-1, 1), vertices=((-5,-5), (-5, 5), (5,5), (5, -5)), axis=axis)
> 
>     # assert that the planes `axis=3`, `axis=-3` (eg. x=3, x=-3 for axis=0) do not intersect with the polyslab.
>     # assert that the bounds has size  `[10, 10].insert(3, axis)`
>     # assert that the point `[0,0].insert(3, axis)` is not inside the polyslab. 
> ```
> 
> Just to make sure that the geometric primitive is working as expected. For example, we had a bug a few months ago where I forgot to handle the axis correctly for cylinder's inside function.

The test has been added.

**Comment by tylerflex** [comment]
feel free to merge this whenever it is ready.

**Review by xin-flex — APPROVED** [review]

**Pull request merged**